debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/processor/TransactionCommitConsumer.java

```diff
@@ -109,233 +102,383 @@ public void accept(LogMinerEvent event) throws InterruptedException {
-            LobState currentLobState = resolveLobStateByCurrentEvent(event);
-            if (currentLobState != this.lobState) {
-                if (this.lobState == LobState.WRITE) {
-                    mergeLobWriteData(lastEvent);
-                }
-                this.lobState = currentLobState;
-            }
+    private void acceptDmlEvent(DmlEvent event) throws InterruptedException {
+        final Table table = schema.tableFor(event.getTableId());
+        if (table == null) {
+            LOGGER.trace("Unable to locate table '{}' schema, ignoring event.", event.getTableId());
+            return;
+        }
 
-            if (!isMerged(event, lastEvent)) {
-                LOGGER.trace("\tMerged skipped.");
-                // Events were not merged, dispatch last one and cache new
-                dispatchChangeEvent(lastEvent);
-                this.lastEvent = event;
-            }
-            else {
-                LOGGER.trace("\tMerged successfully.");
+        String rowId = rowIdFromEvent(table, event);
+        DmlEvent accumulatorEvent = rows.get(rowId);
+        if (!tryMerge(accumulatorEvent, event)) {
+            prepareAndDispatch(accumulatorEvent);
+            if (rowId.equals(currentLobRowId)) {
+                currentLobRowId = null;
+                currentLobColumnName = null;
             }
+            rows.put(rowId, event);
+            accumulatorEvent = event;
         }
-    }
 
-    private void dispatchChangeEvent(LogMinerEvent event) throws InterruptedException {
-        LOGGER.trace("\tEmitting event {} {}", event.getEventType(), event);
-        delegate.accept(event);
+        if (EventType.SELECT_LOB_LOCATOR == event.getEventType()) {
+            currentLobRowId = rowId;
+            currentLobColumnName = ((SelectLobLocatorEvent) event).getColumnName();
+            currentLobColumnPosition = LogMinerHelper.getColumnIndexByName(currentLobColumnName, table);
+
+            // put a LobUnderConstruction in the accumulating event's newValues
+            Object[] values = newValues(accumulatorEvent);
+            Object prevValue = values[currentLobColumnPosition];
+            values[currentLobColumnPosition] = LobUnderConstruction.fromInitialValue(prevValue);
+        }
     }
 
-    private LobState resolveLobStateByCurrentEvent(LogMinerEvent event) {
-        switch (event.getEventType()) {
-            case LOB_WRITE:
-                return LobState.WRITE;
-            case LOB_ERASE:
-                return LobState.ERASE;
-            default:
-                return LobState.OTHER;
+    private void acceptLobManipulationEvent(LogMinerEvent event) {
+        if (null == currentLobRowId || null == currentLobColumnName) {
+            // should only happen when we start streaming in the middle of a LOB transaction (DBZ-4367)
+            LOGGER.trace("got LOB manipulation event without preceding LOB selector; ignoring {} {}", event.getEventType(), event);
```

Comment:
It's a small nit-pick but could we be consistent with the logging messages such that the first letter of the first word is capitalized?  There are multiple occurrences of this in this PR, so I've only pointed out this one.  Thanks!